core/src/main/scala/kafka/server/AbstractFetcherThread.scala

```diff
@@ -339,8 +345,10 @@ abstract class AbstractFetcherThread(name: String,
                       error(s"Error while processing data for partition $topicPartition", e)
                       partitionsWithError += topicPartition
                     case e: Throwable =>
-                      throw new KafkaException(s"Error processing data for partition $topicPartition " +
-                        s"offset ${currentFetchState.fetchOffset}", e)
+                      // drop this partition from the fetcher thread and store in a set for failed partitions
+                      error(s"Unexpected error occurred while processing data for partition $topicPartition", e)
```

Comment:
By the way, it would be helpful to add the fetch offset to this message as we had in the old exception message.

Reply:
Oh yeah, added it to the error message.